test/instrumentation/modules/@aws-sdk/client-sns.test.js

```diff
@@ -0,0 +1,244 @@
+const { NODE_VER_RANGE_IITM } = require('../../../testconsts');
+
+const LOCALSTACK_HOST = process.env.LOCALSTACK_HOST || 'localhost';
+const endpoint = 'http://' + LOCALSTACK_HOST + ':4566';
+
+const testFixtures = [
+  {
+    name: 'simple SNS V3 usage scenario',
+    script: 'fixtures/use-client-sns.js',
+    cwd: __dirname,
+    timeout: 20000, // sanity guard on the test hanging
+    maxBuffer: 10 * 1024 * 1024, // This is big, but I don't ever want this to be a failure reason.
+    env: {
+      AWS_ACCESS_KEY_ID: 'fake',
+      AWS_SECRET_ACCESS_KEY: 'fake',
+      TEST_TOPIC_NAME: 'elasticapmtest-topic-3',
+      TEST_ENDPOINT: endpoint,
+      TEST_REGION: 'us-east-2',
+    },
+    verbose: false,
+    checkApmServer: (t, apmServer) => {
+      t.ok(apmServer.events[0].metadata, 'metadata');
+      const events = sortApmEvents(apmServer.events);
+
+      // First the transaction.
+      t.ok(events[0].transaction, 'got the transaction');
+      const tx = events.shift().transaction;
+      const errors = events.filter((e) => e.error).map((e) => e.error);
+
+      // Compare some common fields across all spans.
+      // ignore http/external spans
+      const spans = events
+        .filter((e) => e.span && e.span.type !== 'external')
+        .map((e) => e.span);
+      spans.forEach((s) => {
+        const errs = validateSpan(s);
+        t.equal(errs, null, 'span is valid (per apm-server intake schema)');
+      });
+      t.equal(
+        spans.filter((s) => s.trace_id === tx.trace_id).length,
+        spans.length,
+        'all spans have the same trace_id',
+      );
+      t.equal(
+        spans.filter((s) => s.transaction_id === tx.id).length,
+        spans.length,
+        'all spans have the same transaction_id',
+      );
+      t.equal(
+        spans.filter((s) => s.sync === false).length,
+        spans.length,
+        'all spans have sync=false',
+      );
+      t.equal(
+        spans.filter((s) => s.sample_rate === 1).length,
+        spans.length,
+        'all spans have sample_rate=1',
+      );
+      const failingSpanId = spans[2].id; // index of `Publish` to a non exixtent topic
+      spans.forEach((s) => {
+        // Remove variable and common fields to facilitate t.deepEqual below.
+        delete s.id;
+        delete s.transaction_id;
+        delete s.parent_id;
+        delete s.trace_id;
+        delete s.timestamp;
+        delete s.duration;
+        delete s.sync;
+        delete s.sample_rate;
+      });
+
+      t.deepEqual(
+        spans.shift(),
+        {
+          name: 'SNS Publish to <PHONE_NUMBER>',
+          type: 'messaging',
+          subtype: 'sns',
+          action: 'Publish',
+          context: {
+            service: {
+              target: { type: 'sns', name: '<PHONE_NUMBER>' },
+            },
+            destination: {
+              service: { name: '', type: '', resource: 'sns/<PHONE_NUMBER>' },
+              address: 'localhost',
```

Comment:
```suggestion
              address: LOCALSTACK_HOST,
```

I think this will fix the failing test: https://github.com/elastic/apm-agent-nodejs/actions/runs/5730209405/job/15528533811?pr=3532